### ISSUE TITLE
Some problems with parallel test result

### ISSUE THREAD

**Issue by honggyukim** [namhyung/uftrace#616]
As requested in #569, I made a new issue regarding parallel test problems.

Thanks to @DanielTimLee's work but I get many warning messages.
```
118 thread_tsd          : OK OK OK OK OK OK OK OK OK OK
119 malloc_hook         : OK OK OK OK OK OK OK OK OK OK
120 malloc_tsd          : OK OK OK OK OK OK OK OK OK OK
121 malloc_fork         : OK OK OK OK OK OK OK OK OK OK
122 time_range2         : OK OK OK OK OK OK OK OK OK OK
123 backtrace           : OK OK OK OK OK OK OK OK OK OK
WARN: failed to mmap perf event: Operation not permitted
WARN: failed to mmap perf event: Operation not permitted
WARN: failed to mmap perf event: Operation not permitted
WARN: failed to mmap perf event: Operation not permitted
WARN: failed to mmap perf event: Operation not permitted
WARN: failed to mmap perf event: Operation not permitted
124 exception           : OK OK OK OK OK OK OK OK OK OK
125 report_range        : OK OK OK OK OK OK OK OK OK OK
126 arg_regex           : OK OK OK OK OK SK SK SK SK SK
127 arg_module          : OK OK OK OK OK SK SK SK SK SK
128 arg_module2         : OK OK OK OK OK SK SK SK SK SK
129 session_tid         : OK OK OK OK OK OK OK OK OK OK
WARN: failed to mmap perf event: Operation not permitted
WARN: failed to mmap perf event: Operation not permitted
WARN: failed to mmap perf event: Operation not permitted
130 thread_exec         : OK OK OK OK OK OK OK OK OK OK
131 lib_dlopen          : OK OK OK OK OK OK OK OK OK OK
132 trigger_kernel      : SK SK SK SK SK SK SK SK SK SK
```
And some of the result results are randomly changed.
```
$ ./runtest.py
        ...
158 report_diff_policy1 : OK OK OK OK NG OK OK OK OK OK
159 report_diff_policy2 : OK OK OK OK OK OK OK OK OK OK
160 report_diff_policy3 : OK OK OK OK OK OK OK OK OK OK
161 pltbind_now         : NG NG NG NG NG NG NG NG NG NG
162 pltbind_now_pie     : BI BI BI BI BI BI BI BI BI BI
163 event_sched         : OK NG NG NG NG NG NG NG OK NG
164 report_sched        : OK OK NG NG NG NG NG OK OK OK
165 graph_sched         : OK NG OK OK OK OK NG NG NG OK
166 dump_sched          : OK OK OK OK OK OK OK OK OK OK
167 recv_sched          : OK NG OK OK OK NG OK OK OK OK
168 lib_nested          : OK OK OK OK OK OK OK OK OK OK
        ...
```
Many of 163 event_sched are failed but the result is different if I run it as a single test.
```
$ ./runtest.py 163
Start 1 tests with 40 worker
Test case                 pg             finstrument-fu
------------------------: O0 O1 O2 O3 Os O0 O1 O2 O3 Os
163 event_sched         : OK OK OK OK OK OK OK OK OK OK
```
The final result shows that many tests are failed anyway.
```
runtime test stats
====================
total  2120  Tests executed (success: 81.65%)
  OK:  1681  Test succeeded
  OK:    50  Test succeeded (with some fixup)
  NG:    72  Different test result
  NZ:     0  Non-zero return value
  SG:     2  Abnormal exit by signal
  TM:     0  Test ran too long
  BI:    20  Build failed
  LA:     0  Unsupported Language
  SK:   295  Skipped
```
We may have to fix some more sync issues.

**Comment by honggyukim** [comment]
Here is my system stat info.
```
$ ulimit -a | grep locked
max locked memory       (kbytes, -l) 64

$ grep locked /proc/meminfo
Mlocked:            3652 kB
```

**Comment by namhyung** [comment]
You should either increase the mlock limit (`ulimit -l`) or write "-1" to `/proc/sys/kernel/perf_event_paranoid`.

**Comment by honggyukim** [comment]
Hmm.. why don't we better to make some throttle on the number of parallel jobs. I think it's better to be working on the default settings.

**Comment by namhyung** [comment]
I guess it's changing dynamically.  From the numbers you wrote above, single run will also fail IMO.

**Comment by honggyukim** [comment]
Thanks. It works fine if I change `/proc/sys/kernel/perf_event_paranoid` to `-1`. But I think it's better to be working with default environment, so tried again with `./runtest.py -j8` and it works fine as well.

So I think it's better to have a way to set the number of workers from user commands. If we directly run `runtest.py` then it's fine, and it can be written as following:
```
$ make test TESTARG=-j8
```
For better usage, I would like to make it work with the following command
```
$ make test -j8
```
But I don't know how the detect the current number of jobs inside `Makefile` yet. If it's possible, I think default parallel jobs have to be 1 instead of the number of cores.

**Comment by DanielTimLee** [comment]
Well, we could add some logic to parse argument and pass to `readtest.py`, in `Makefile` test target.

**Comment by honggyukim** [comment]
I couldn't find a way to do that yet. Please let me know if it's possible. Thanks for your great work anyway!

**Comment by honggyukim** [comment]
That could be a way. @namhyung what do you think?

**Comment by namhyung** [comment]
Maybe you could parse `MAKEFLAGS` and set `TESTARG`.  You might want to take a look at the Makefile of perf tools.

https://elixir.bootlin.com/linux/latest/source/tools/perf/Makefile

**Comment by DanielTimLee** [comment]
Maybe like this would be available to pass an argument with `MAKEFLAGS`?
#618

**Comment by namhyung** [comment]
> I think default parallel jobs have to be 1 instead of the number of cores.

@honggyukim I don't think it's a good default.  We prefer jobs done quickly! :)  Ideally we would make some part serialized if needed.  For example, checking mlock limit before running cases and run perf event (sched) test cases serially.  Also kernel tests always need to be run serially.

But anyway we can control it by `-j` option so I'd leave it as a later work.

**Comment by namhyung** [comment]
I guess this issue was resolved by #698
